pgjdbc/src/main/java/org/postgresql/core/SqlCommandType.java

```diff
@@ -25,5 +27,13 @@ public enum SqlCommandType {
   SELECT,
   WITH,
   CREATE,
-  ALTER;
+  ALTER,
+  SET;
+
+  public static SqlCommandType fromStatus(String value) {
+    return Arrays.stream(SqlCommandType.values())
+        .filter(v -> v.name().equalsIgnoreCase(value))
+        .findFirst()
+        .orElse(BLANK);
+  }
```

Comment:
`SqlCommandType.values()` allocates the array for every execution, and `Arrays.stream` does not seem to be that efficient either.

WDYT of caching `.values()` and using a regular `for` loop here?

Reply:
EnumSet is now used